components/script/dom/webglsampler.rs

```diff
@@ -0,0 +1,187 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::dom::bindings::codegen::Bindings::WebGL2RenderingContextBinding::WebGL2RenderingContextConstants as constants;
+use crate::dom::bindings::codegen::Bindings::WebGLSamplerBinding;
+use crate::dom::bindings::inheritance::Castable;
+use crate::dom::bindings::reflector::{reflect_dom_object, DomObject};
+use crate::dom::bindings::root::DomRoot;
+use crate::dom::webglobject::WebGLObject;
+use crate::dom::webglrenderingcontext::WebGLRenderingContext;
+use canvas_traits::webgl::WebGLError::*;
+use canvas_traits::webgl::{webgl_channel, WebGLCommand, WebGLSamplerId};
+use dom_struct::dom_struct;
+use std::cell::Cell;
+
+#[dom_struct]
+pub struct WebGLSampler {
+    webgl_object: WebGLObject,
+    gl_id: WebGLSamplerId,
+    marked_for_deletion: Cell<bool>,
+}
+
+#[derive(Clone, Copy)]
+pub enum WebGLSamplerValue {
+    Float(f32),
+    GLenum(u32),
+}
+
+fn validate_params(pname: u32, value: WebGLSamplerValue) -> bool {
+    match value {
+        WebGLSamplerValue::GLenum(value) => {
+            let allowed_values = match pname {
+                constants::TEXTURE_MIN_FILTER => vec![
```

Comment:
Yes, that's what I tried first, but the compiler refuses it because different branches return slices with different number of elements. Perhaps we should use something like SmallVec?

Reply:
Use &[FOO][..] to force slices instead of sized arrays.